src/analysis/functions.rs

```diff
@@ -238,51 +260,201 @@ fn analyze_function(
+                assert!(
+                    !par.instance_parameter || pos == 0,
+                    "Wrong instance parameter in {}",
+                    func.c_identifier.as_ref().unwrap()
+                );
+                if let Ok(s) = used_rust_type(env, par.typ, !par.direction.is_out()) {
+                    used_types.push(s);
+                }
+                let (to_glib_extra, callback_info) = bounds.add_for_parameter(env, func, par, async, expecting_data);
+                if let Some(to_glib_extra) = to_glib_extra {
+                    if par.c_type != "GDestroyNotify" {
+                        to_glib_extras.insert(pos, to_glib_extra);
+                    }
+                }
+
+                let func_name = match &func.c_identifier {
+                    Some(n) => &n,
+                    None => &func.name,
+                };
+                if par.c_type.ends_with("Func") || par.c_type.ends_with("Callback") {
+                    if let Some((mut callback, destroy_index)) = analyze_callback(
+                        //&func.name,
+                        func_name,
+                        env,
+                        &par,
+                        &callback_info,
+                        &mut commented,
+                        imports,
+                        &c_parameters,
+                    ) {
+                        if let Some(destroy_index) = destroy_index {
+                            let user_data = cross_user_data_check.entry(destroy_index)
+                                                                 .or_insert_with(|| callback.user_data_index);
+                            if *user_data != callback.user_data_index {
+                                error!("`{}`: Different destructors cannot share the same user data",
+                                       func.name);
+                                commented = true;
+                            }
+                            callback.destroy_index = destroy_index;
+                        } else {
+                            user_data_indexes.insert(callback.user_data_index);
+                            to_remove.push(callback.user_data_index);
+                        }
+                        callbacks.push(callback);
+                        to_replace.push((pos, par.typ));
+                        continue;
+                    }
+                } else if par.c_type == "GDestroyNotify" {
+                    if let Some((mut callback, _)) = analyze_callback(
+                        func_name,
+                        env,
+                        &par,
+                        &callback_info,
+                        &mut commented,
+                        imports,
+                        &c_parameters,
+                    ) {
+                        // We just assume that for API "cleaness", the destroy callback will always be
+                        // |-> *after* <-| the initial callback.
```

Comment:
Should be safe, but if this is not the case we go into the `error!` below? Sounds fine